### PR TITLE
Add configuration_internal to build-ami script

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] Have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?

### CODE REVIEW

**Comment by nadeemshahzad** [comment]
@edx/devops Please review

File: util/jenkins/build-ami.sh [hunk 1]
```diff
@@ -64,6 +64,10 @@ cd $WORKSPACE/configuration-secure
 configuration_secure=`git rev-parse --short HEAD`
 cd $WORKSPACE
 
+cd $WORKSPACE/configuration-internal
+configuration_internal=`git rev-parse --short HEAD`
```
Comment (feanil) [inline util/jenkins/build-ami.sh]: @nadeemshahzad this variable is never again used in this file, how is it propagated to abbey.py which launches the remote instance?

Comment (nadeemshahzad) [reply util/jenkins/build-ami.sh]: @feanil Adding configuration internal Git SCM to job and 
```export configurationinternalversion="$configuration_internal" ``` in shell its value will be passed to abbey.py or should I set its value here in file?
```
if [[ ! -z "$configurationinternalrepo" ]]; then
  configurationinternal_params="--configuration-internal-repo $configurationinternalrepo"
  if [[ ! -z "$configurationinternalversion" ]]; then
    configurationinternal_params="$configurationinternal_params --configuration-internal-version $configurationinternalversion"
  fi
fi
```

**Comment by nadeemshahzad** [comment]
@feanil I have added configuration_internal var to file and updated "abbey.py" to tag image with configuration_internal repository and version.

**Comment by nadeemshahzad** [comment]
@feanil I have updated the PR to get configuration_internal_version from vars and passed its value to abbey.py this will make alton to work with config_internal_version.

**Comment by feanil** [comment]
The problem with doing it this way is that in the alton command, we'll need to refer to configuration, and configuration_secure without the 'version' suffix but configuration_internal will have a version suffix.  @jibsheet @fredsmith @jdmulloy you will be the primary consumers of this, so you should decide if that cognitive dissonance will cause problems or if it's fine.  eg.

# correct alton command
`@alton cut ami for prod-edx-edxapp from prod-edx-edxapp with configuration=master configuration_internal_version=master`

# incorrect alton command
`@alton cut ami for prod-edx-edxapp from prod-edx-edxapp with configuration=master configuration_internal=master`

Note that the second one looks right based on the mental heuristic we use for configuration and configuration_secure(no `_version` suffix) but is in fact incorrect.

**Comment by jibsheet** [comment]
I can live with configuration_internal_version as long as we update the docs https://openedx.atlassian.net/wiki/spaces/EdxOps/pages/35422870/How+to+Use+Alton (look for configuration_secure and add some internal examples)

Unfortunately, this tags with the branch name, rather than unrolling that to a sha1.  Can we unroll that in the metadata script once we have the clone (using the git rev-parse HEAD that build-ami.sh has).  This way if I build two amis with configuration_internal_version=jibsheet/test I can tell the difference, since they'll be tagged with sha1s, not the same branch name.  I think it's just a simple export here https://github.com/edx/configuration/blob/31e113b8862d49f9db1b8332a01175338e8bd301/util/vpc-tools/abbey.py#L496 and using it here https://github.com/edx/configuration/blob/31e113b8862d49f9db1b8332a01175338e8bd301/util/vpc-tools/abbey.py#L720

I realize we also apparently don't tag the material in gocd, but that's a separate ticket.

**Comment by feanil** [comment]
@jibsheet The first place you pointed to is a bash script that runs in the userdata on the new machine.  The second place you pointed to runs on Jenkins.  The only mechanism we have to pass from A to B is to add a tag to the image and then grab it via this block. https://github.com/edx/configuration/blob/31e113b8862d49f9db1b8332a01175338e8bd301/util/vpc-tools/abbey.py#L727-L731

**Comment by jibsheet** [comment]
@feanil ah, yes.  sigh.  I guess we live with the bug and try to prioritize test GoCD pipelines.  Let's make sure the docs are up to date at least.

**Comment by feanil** [comment]
Agreed.  @nadeemshahzad please update tho documentation to show how to use the new variable for controlling the version of configuration_internal.  Once that is done and reviewed, I think this can be merged.

**Comment by nadeemshahzad** [comment]
@feanil I have updated the alton documentation for 'configuration_internal_version' variable.
https://openedx.atlassian.net/wiki/spaces/EdxOps/pages/35422870/How+to+Use+Alton

**Review by feanil — APPROVED** [review]

**Pull request merged**